src/main/java/com/code_intelligence/jazzer/mutation/combinator/MutatorCombinators.java

```diff
@@ -0,0 +1,130 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.jazzer.mutation.combinator;
+
+import static com.code_intelligence.jazzer.mutation.support.Preconditions.require;
+import static com.code_intelligence.jazzer.mutation.support.Preconditions.requireNonNullElements;
+import static java.util.Arrays.stream;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+import com.code_intelligence.jazzer.mutation.api.InPlaceMutator;
+import com.code_intelligence.jazzer.mutation.api.PseudoRandom;
+import com.code_intelligence.jazzer.mutation.api.SerializingMutator;
+import com.code_intelligence.jazzer.mutation.api.ValueMutator;
+import java.util.Arrays;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import net.jodah.typetools.TypeResolver;
+
+public final class MutatorCombinators {
+  private MutatorCombinators() {}
+
+  public static <T, R> InPlaceMutator<T> mutateProperty(
+      Function<T, R> getter, ValueMutator<R> mutator, BiConsumer<T, R> setter) {
+    requireNonNull(getter);
+    requireNonNull(mutator);
+    requireNonNull(setter);
+    return new InPlaceMutator<T>() {
+      @Override
+      public void initInPlace(T reference, PseudoRandom prng) {
+        setter.accept(reference, mutator.init(prng));
+      }
+
+      @Override
+      public void mutateInPlace(T reference, PseudoRandom prng) {
+        setter.accept(reference, mutator.mutate(getter.apply(reference), prng));
+      }
+
+      @Override
+      public String toString() {
+        Class<?> owningType =
+            TypeResolver.resolveRawArguments(Function.class, getter.getClass())[0];
+        return owningType.getSimpleName() + "." + mutator;
+      }
+    };
+  }
+
+  public static <T, R> InPlaceMutator<T> mutateViaView(
+      Function<T, R> map, InPlaceMutator<R> mutator) {
+    requireNonNull(map);
+    requireNonNull(mutator);
+    return new InPlaceMutator<T>() {
+      @Override
+      public void initInPlace(T reference, PseudoRandom prng) {
+        mutator.initInPlace(map.apply(reference), prng);
+      }
+
+      @Override
+      public void mutateInPlace(T reference, PseudoRandom prng) {
+        mutator.mutateInPlace(map.apply(reference), prng);
+      }
+
+      @Override
+      public String toString() {
+        Class<?> owningType = TypeResolver.resolveRawArguments(Function.class, map.getClass())[0];
+        return owningType.getSimpleName() + " via " + mutator;
+      }
+    };
+  }
+
+  @SafeVarargs
+  public static <T> InPlaceMutator<T> combine(InPlaceMutator<T>... partialMutators) {
```

Comment:
How does this differ from `ProductMutator`?

Reply:
Added comments to both functions that I hope explain the difference (summary: it's the type, which is T in one case and Object[] in the other).

Reply:
Thx for the clarification. I missed its usage in combination with `mutateView` and `mutateProperty`.